### PR TITLE
Fixes #17525 - Add a friendly error message to invalid dns_id

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

**Comment by shlomizadok** [comment]
From https://github.com/rails/rails/pull/22236 I learn there is no support in foreign key constraint in SQLite3 adapter (rails 4.2).
@domcleal any advise? to run the [assertion](https://github.com/theforeman/foreman/pull/4120/files#diff-93dedd06e33e0eda2f2e91295fa6be34R33) only on pg/mysql? Thanks :)

**Comment by shlomizadok** [comment]
By the way, the domain will be saved on sqlite3, even with invalid dns_id

File: test/controllers/api/v2/domains_controller_test.rb [hunk 1]
```diff
@@ -26,6 +26,18 @@ class Api::V2::DomainsControllerTest < ActionController::TestCase
     assert_response :unprocessable_entity
   end
 
+  test "should not create invalid dns_id" do
+    # Currently Rails 4.2 does not support foreign key constraint with sqlite3
+    # (See https://github.com/rails/rails/pull/22236)
+    # Skipping this test until resolved
+    skip if ActiveRecord::Base.connection_config[:adapter].eql?"sqlite3"
+    invalid_proxy_id = SmartProxy.last.id + 100
+    post :create, { :domain => { :name => "doma.in", :dns_id => invalid_proxy_id } }
```
[the remark this answers is not in the document]
Comment (shlomizadok) [reply test/controllers/api/v2/domains_controller_test.rb:35]: woof!

**Comment by dLobatog** [comment]
Thanks @shlomizadok !

**Pull request merged**